swift-sdk/Internal/AnonymousUserManager.swift

```diff
@@ -204,6 +205,11 @@ public class AnonymousUserManager: AnonymousUserManagerProtocol {
         } else {
             eventsDataObjects.append(appendData)
         }
+
+        let eventDataCount = eventsDataObjects.count
+        if  eventDataCount > config.eventThresholdLimit {
+            eventsDataObjects = eventsDataObjects.suffix(config.eventThresholdLimit)
```

Comment:
Right now the number of events stored is one more than the threshold passed in on the config.

Reply:
I will update this and merge.

Reply:
Nevermind this works.